packages/account/src/providers/transaction-summary/input.ts

```diff
@@ -33,16 +34,23 @@ export function getInputsContract(inputs: Input[]) {
 }
 
 /** @hidden */
-export function getInputFromAssetId(inputs: Input[], assetId: string) {
+export function getRelevantInputs(inputs: Input[], assetId: string = ZeroBytes32) {
   const coinInputs = getInputsCoin(inputs);
   const messageInputs = getInputsMessage(inputs);
-  const coinInput = coinInputs.find((i) => i.assetId === assetId);
-  // TODO: should include assetId in InputMessage as well. for now we're mocking ETH
-  const messageInput = messageInputs.find(
-    (_) => assetId === '0x0000000000000000000000000000000000000000000000000000000000000000'
-  );
+  let coinInput = coinInputs.find((i) => i.assetId === assetId);
+  // #TODO: There are times when the ReceiptCall's baseAssetId doesn't match the CoinInput's assetId
+  // In this case, we should return the last input in the CoinInput array
+  if (!coinInput && coinInputs.length > 0) {
+    coinInput = coinInputs[coinInputs.length - 1];
+  }
+
+  const messageInput = messageInputs.find(({ amount }) => !!amount && amount.gt(0));
+
+  if (coinInput) {
+    return coinInput;
+  }
 
-  return coinInput || messageInput;
+  return messageInput;
 }
```

Comment:
3. It is better to change the name of this helper later on when we decide to refactor all these helpers.

This helper was created to extract the first input from the input array with the provided `assetId`. We can only return an `InputMessage` here if the provided `assetId` is the base asset ID

So we can change it to this (feel free to change the logic but the behavior needs to be the same)
```ts
export function getInputFromAssetId(inputs: Input[], assetId: string, isBaseAsset = false) {
  const coinInputs = getInputsCoin(inputs);
  const coinInput = coinInputs.find((i) => i.assetId === assetId);

  let messageInput: InputMessage | undefined;
  if (isBaseAsset) {
    messageInput = getInputsMessage(inputs)?.[0];
  }

  return coinInput || messageInput;
}
```

Reply:
7004319a6